akka-http-scalafix/scalafix-rules/src/main/scala/akka/http/fix/MigrateToServerBuilder.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.fix
+
+import scalafix.lint.LintSeverity
+import scalafix.v1._
+
+import scala.meta._
+import scala.util.Try
+
+class MigrateToServerBuilder extends SemanticRule("MigrateToServerBuilder") {
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    def patch(t: Tree, http: Term, targetMethod: Term => String): Patch = {
+      val args = t.symbol.info.get.signature.asInstanceOf[MethodSignature].parameterLists.head.map(_.displayName)
+      val materializerAndTarget: Option[(Tree, Term)] =
+        Try {
+          val sig = t.parent.get.symbol.info.get.signature.asInstanceOf[MethodSignature]
+          require(sig.parameterLists(1)(0).signature.toString == "Materializer")
+          (t.parent.get, t.parent.get.asInstanceOf[Term.Apply].args.head)
+        }.toOption
+
+      val materializerLint: Option[Patch] = materializerAndTarget.map {
+        case (_, matArg) =>
+          Patch.lint(Diagnostic(
+            "custom-materializer-warning",
+            "Custom materializers are often not needed any more. You can often just remove custom materializers and " +
+              "just use the system materializer which is supplied automatically.",
+            matArg.pos,
+            severity = LintSeverity.Warning
+          ))
+      }
+
+      val argExps = namedArgMap(args, t.asInstanceOf[Term.Apply].args) ++ materializerAndTarget.map("materializer" -> _._2).toSeq
+      val targetTree = materializerAndTarget.map(_._1).getOrElse(t) // patch parent if materializer arg is found
+
+      patchTree(targetTree, http, argExps, targetMethod(argExps("handler"))) + materializerLint
+    }
+
+    def patchTree(t: Tree, http: Term, argExps: Map[String, Term], targetMethod: String): Patch =
+      Patch.replaceTree(t, s"${builder(http, argExps)}.$targetMethod(${argExps("handler")})")
+
+    def handlerIsRoute(handler: Term): Boolean =
+      handler.symbol.info.exists(_.signature.toString contains "Route") || // doesn't seem to work with synthetics on
+        (handler.synthetics match { // only works with `scalacOptions += "-P:semanticdb:synthetics:on"`
+          case ApplyTree(fun, _) :: Nil => fun.symbol.exists(_.displayName == "routeToFlow") /* FIXME: could use real symbol instead */
```

Comment:
Want to clear the FIXME?

Reply:
Clarifying the comment instead:

```suggestion
          case ApplyTree(fun, _) :: Nil => fun.symbol.exists(_.displayName == "routeToFlow") // somewhat inaccurate, but that name should be unique enough for our purposes
```